api/v1alpha1/ext_auth_types.go

```diff
@@ -59,39 +59,24 @@ type ExtAuth struct {
 // The authorization request message is defined in
 // https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto
 type GRPCExtAuthService struct {
-	// Host is the hostname of the gRPC External Authorization service.
-	Host gwapiv1a2.PreciseHostname `json:"host"`
-
-	// Port is the network port of the gRPC External Authorization service.
-	Port gwapiv1a2.PortNumber `json:"port"`
-
-	// TLS defines the TLS configuration for the gRPC External Authorization service.
-	// Note: If not specified, the proxy will talk to the gRPC External Authorization
-	// service in plaintext.
-	// +optional
-	TLS *TLSConfig `json:"tls,omitempty"`
+	// BackendObjectReference references a Kubernetes object that represents the
```

Comment:
Should this live in the top level ?

Reply:
It can be changed to the below, I slightly prefer the current one.

```
type ExtAuth struct {
	// Type decides the type of External Authorization.
	// Valid ExtAuthServiceType values are "GRPC" or "HTTP".
	// +kubebuilder:validation:Enum=GRPC;HTTP
	// +unionDiscriminator
	Type ExtAuthServiceType `json:"type"`

	HTTP gwapiv1.BackendObjectReference

	GRPC gwapiv1.BackendObjectReference
	
	HeadersToExtAuth []string `json:"headersToExtAuth,omitempty"`
		
	// The below configurations only apply to HTTP external authorization service.
	// More will be added in the future.
	Path *string `json:"path,omitempty"`

	HeadersToBackend []string `json:"headersToBackend,omitempty"`
}
```

Reply:
why do we need a unique field for HTTP and GRPC backends ?